### PR TITLE
Quic support part 1

### PR DESCRIPTION
**Pull request by qzhuyan**
green build on ubuntu/centos/debian

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/emqx/emqx/blob/master/CONTRIBUTING.md.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

**Review by turtleDeng — APPROVED** [review]

File: apps/emqx/src/emqx_schema.erl [hunk 1]
```diff
@@ -356,6 +360,31 @@ fields("wss_listener_settings") ->
...
+                  , "buffer"
+                  , "high_watermark"
+                  , "tune_buffer"
+                  , "nodelay"
+                  , "reuseaddr"
+                  ],
+    lists:foldl(fun(K, Acc) ->
+                        lists:keydelete(K, 1, Acc)
+                end,
+                [ {"certfile", t(string(), "emqx.certfile", undefined)}
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:382]: `emqx.certfile` is too general, better to be `emqx.quic.certfile` ?

File: apps/emqx/src/emqx_quic_stream.erl [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        {error, _} -> {error, closed};
+        Res -> Res
+    end.
+
+setopts(_Socket, _Opts) ->
+    ok.
+
+getopts(_Socket, _Opts) ->
+    %% todo
+    { ok, [{high_watermark, 0},
```
Comment (k32) [inline apps/emqx/src/emqx_quic_stream.erl:60]: Nit: extra space before ok.

File: apps/emqx/src/emqx_quic_stream.erl [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+-spec(ensure_ok_or_exit(atom(), list(term())) -> term()).
+ensure_ok_or_exit(Fun, Args = [Sock|_]) when is_atom(Fun), is_list(Args) ->
+    case erlang:apply(?MODULE, Fun, Args) of
+        {error, Reason} when Reason =:= enotconn; Reason =:= closed ->
+            fast_close(Sock),
+            exit(normal);
+        {error, Reason} ->
+            fast_close(Sock),
+            exit({shutdown, Reason});
+         Result -> Result
```
Comment (k32) [inline apps/emqx/src/emqx_quic_stream.erl:80]: Nit: indentation

**Comment by k32** [comment]
LGTM, but someone else should approve

File: apps/emqx_auth_mnesia/src/emqx_auth_mnesia.app.src [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {application, emqx_auth_mnesia,
  [{description, "EMQ X Authentication with Mnesia"},
-  {vsn, "4.3.0"}, % strict semver, bump manually
+  {vsn, "5.0.0"}, % strict semver, bump manually
```
Comment (zmstone) [inline apps/emqx_auth_mnesia/src/emqx_auth_mnesia.app.src:3]: we have decided not to keep app version up with release version.

**Review by k32 — APPROVED** [review]

**Review by turtleDeng — APPROVED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**